The code quality test has failed because of trailing whitespace:
https://github.com/sympy/sympy/actions/runs/10746113244/job/29806427330?pr=27043

The error message tells you which file and which line:

> FAILED sympy/testing/tests/test_code_quality.py::test_files - AssertionError: File contains trailing whitespace: /home/runner/work/sympy/sympy/sympy/printing/str.py, line 741.

You can delete the spaces at the end of the line or run the `bin/strip_whitespace` script. Once that is done you can push another commit to your branch which will update the pull request.

You do *not* need to close the pull request and open a new one.

--- a/.mailmap
+++ b/.mailmap
@@ -1221,6 +1221,7 @@ Riley Britten <nrb1324@hotmail.com>
 Rimi <rimibis@umich.edu>
 Rishabh Daal <rishabhdaal@gmail.com>
 Rishabh Dixit <rishabhdixit11@gmail.com>
+Rishabh Kamboj <111004091+VectorNd@users.noreply.github.com> VectorNd <rishabh.kamboj.eee21@itbhu.ac.in>
 Rishabh Madan <rishabhmadan96@gmail.com>
 Rishat Iskhakov <iskhakov@frtk.ru>
 Rishav Chakraborty <annonymousxyz@outlook.com>

--- a/sympy/geometry/line.py
+++ b/sympy/geometry/line.py
@@ -2587,7 +2587,7 @@ class Line3D(LinearEntity3D, Line):
         # eliminate k from equations by solving first eq with k for k
         for i, e in enumerate(eqs):
             if e.has(k):
-                kk = solve(eqs[i], k)[0]
+                kk = solve(e, k)[0]
                 eqs.pop(i)
                 break
         return Tuple(*[i.subs(k, kk).as_numer_denom()[0] for i in eqs])

--- a/sympy/integrals/integrals.py
+++ b/sympy/integrals/integrals.py
@@ -1568,6 +1568,7 @@ def integrate(*args, meijerg=None, conds='piecewise', risch=None, heurisch=None,
         'heurisch': heurisch,
         'manual': manual
         }
+
     integral = Integral(*args, **kwargs)
 
     if isinstance(integral, Integral):
@@ -1576,7 +1577,6 @@ def integrate(*args, meijerg=None, conds='piecewise', risch=None, heurisch=None,
         new_args = [a.doit(**doit_flags) if isinstance(a, Integral) else a
             for a in integral.args]
         return integral.func(*new_args)
-
 
 def line_integrate(field, curve, vars):
     """line_integrate(field, Curve, variables)

--- a/sympy/physics/vector/dyadic.py
+++ b/sympy/physics/vector/dyadic.py
@@ -173,28 +173,28 @@ class Dyadic(Printable, EvalfMixin):
         ol = []  # output list, to be concatenated to a string
         for i, v in enumerate(ar):
             # if the coef of the dyadic is 1, we skip the 1
-            if ar[i][0] == 1:
-                ol.append(' + ' + printer._print(ar[i][1]) + r"\otimes " +
-                          printer._print(ar[i][2]))
+            if v[0] == 1:
+                ol.append(' + ' + printer._print(v[1]) + r"\otimes " +
+                          printer._print(v[2]))
             # if the coef of the dyadic is -1, we skip the 1
-            elif ar[i][0] == -1:
+            elif v[0] == -1:
                 ol.append(' - ' +
-                          printer._print(ar[i][1]) +
+                          printer._print(v[1]) +
                           r"\otimes " +
-                          printer._print(ar[i][2]))
+                          printer._print(v[2]))
             # If the coefficient of the dyadic is not 1 or -1,
             # we might wrap it in parentheses, for readability.
-            elif ar[i][0] != 0:
-                arg_str = printer._print(ar[i][0])
-                if isinstance(ar[i][0], Add):
+            elif v[0] != 0:
+                arg_str = printer._print(v[0])
+                if isinstance(v[0], Add):
                     arg_str = '(%s)' % arg_str
                 if arg_str.startswith('-'):
                     arg_str = arg_str[1:]
                     str_start = ' - '
                 else:
                     str_start = ' + '
-                ol.append(str_start + arg_str + printer._print(ar[i][1]) +
-                          r"\otimes " + printer._print(ar[i][2]))
+                ol.append(str_start + arg_str + printer._print(v[1]) +
+                          r"\otimes " + printer._print(v[2]))
         outstr = ''.join(ol)
         if outstr.startswith(' + '):
             outstr = outstr[3:]
@@ -217,36 +217,36 @@ class Dyadic(Printable, EvalfMixin):
                 ol = []  # output list, to be concatenated to a string
                 for i, v in enumerate(ar):
                     # if the coef of the dyadic is 1, we skip the 1
-                    if ar[i][0] == 1:
+                    if v[0] == 1:
                         ol.extend([" + ",
-                                  mpp.doprint(ar[i][1]),
+                                  mpp.doprint(v[1]),
                                   bar,
-                                  mpp.doprint(ar[i][2])])
+                                  mpp.doprint(v[2])])
 
                     # if the coef of the dyadic is -1, we skip the 1
-                    elif ar[i][0] == -1:
+                    elif v[0] == -1:
                         ol.extend([" - ",
-                                  mpp.doprint(ar[i][1]),
+                                  mpp.doprint(v[1]),
                                   bar,
-                                  mpp.doprint(ar[i][2])])
+                                  mpp.doprint(v[2])])
 
                     # If the coefficient of the dyadic is not 1 or -1,
                     # we might wrap it in parentheses, for readability.
-                    elif ar[i][0] != 0:
-                        if isinstance(ar[i][0], Add):
+                    elif v[0] != 0:
+                        if isinstance(v[0], Add):
                             arg_str = mpp._print(
-                                ar[i][0]).parens()[0]
+                                v[0]).parens()[0]
                         else:
-                            arg_str = mpp.doprint(ar[i][0])
+                            arg_str = mpp.doprint(v[0])
                         if arg_str.startswith("-"):
                             arg_str = arg_str[1:]
                             str_start = " - "
                         else:
                             str_start = " + "
                         ol.extend([str_start, arg_str, " ",
-                                  mpp.doprint(ar[i][1]),
+                                  mpp.doprint(v[1]),
                                   bar,
-                                  mpp.doprint(ar[i][2])])
+                                  mpp.doprint(v[2])])
 
                 outstr = "".join(ol)
                 if outstr.startswith(" + "):
@@ -267,18 +267,18 @@ class Dyadic(Printable, EvalfMixin):
         ol = []  # output list, to be concatenated to a string
         for i, v in enumerate(ar):
             # if the coef of the dyadic is 1, we skip the 1
-            if ar[i][0] == 1:
-                ol.append(' + (' + printer._print(ar[i][1]) + '|' +
-                          printer._print(ar[i][2]) + ')')
+            if v[0] == 1:
+                ol.append(' + (' + printer._print(v[1]) + '|' +
+                          printer._print(v[2]) + ')')
             # if the coef of the dyadic is -1, we skip the 1
-            elif ar[i][0] == -1:
-                ol.append(' - (' + printer._print(ar[i][1]) + '|' +
-                          printer._print(ar[i][2]) + ')')
+            elif v[0] == -1:
+                ol.append(' - (' + printer._print(v[1]) + '|' +
+                          printer._print(v[2]) + ')')
             # If the coefficient of the dyadic is not 1 or -1,
             # we might wrap it in parentheses, for readability.
-            elif ar[i][0] != 0:
-                arg_str = printer._print(ar[i][0])
-                if isinstance(ar[i][0], Add):
+            elif v[0] != 0:
+                arg_str = printer._print(v[0])
+                if isinstance(v[0], Add):
                     arg_str = "(%s)" % arg_str
                 if arg_str[0] == '-':
                     arg_str = arg_str[1:]
@@ -286,8 +286,8 @@ class Dyadic(Printable, EvalfMixin):
                 else:
                     str_start = ' + '
                 ol.append(str_start + arg_str + '*(' +
-                          printer._print(ar[i][1]) +
-                          '|' + printer._print(ar[i][2]) + ')')
+                          printer._print(v[1]) +
+                          '|' + printer._print(v[2]) + ')')
         outstr = ''.join(ol)
         if outstr.startswith(' + '):
             outstr = outstr[3:]

--- a/sympy/physics/vector/vector.py
+++ b/sympy/physics/vector/vector.py
@@ -231,23 +231,23 @@ class Vector(Printable, EvalfMixin):
         for i, v in enumerate(ar):
             for j in 0, 1, 2:
                 # if the coef of the basis vector is 1, we skip the 1
-                if ar[i][0][j] == 1:
-                    ol.append(' + ' + ar[i][1].latex_vecs[j])
+                if v[0][j] == 1:
+                    ol.append(' + ' + v[1].latex_vecs[j])
                 # if the coef of the basis vector is -1, we skip the 1
-                elif ar[i][0][j] == -1:
-                    ol.append(' - ' + ar[i][1].latex_vecs[j])
-                elif ar[i][0][j] != 0:
+                elif v[0][j] == -1:
+                    ol.append(' - ' + v[1].latex_vecs[j])
+                elif v[0][j] != 0:
                     # If the coefficient of the basis vector is not 1 or -1;
                     # also, we might wrap it in parentheses, for readability.
-                    arg_str = printer._print(ar[i][0][j])
-                    if isinstance(ar[i][0][j], Add):
+                    arg_str = printer._print(v[0][j])
+                    if isinstance(v[0][j], Add):
                         arg_str = "(%s)" % arg_str
                     if arg_str[0] == '-':
                         arg_str = arg_str[1:]
                         str_start = ' - '
                     else:
                         str_start = ' + '
-                    ol.append(str_start + arg_str + ar[i][1].latex_vecs[j])
+                    ol.append(str_start + arg_str + v[1].latex_vecs[j])
         outstr = ''.join(ol)
         if outstr.startswith(' + '):
             outstr = outstr[3:]
@@ -305,23 +305,23 @@ class Vector(Printable, EvalfMixin):
         for i, v in enumerate(ar):
             for j in 0, 1, 2:
                 # if the coef of the basis vector is 1, we skip the 1
-                if ar[i][0][j] == 1:
-                    ol.append(' + ' + ar[i][1].str_vecs[j])
+                if v[0][j] == 1:
+                    ol.append(' + ' + v[1].str_vecs[j])
                 # if the coef of the basis vector is -1, we skip the 1
-                elif ar[i][0][j] == -1:
-                    ol.append(' - ' + ar[i][1].str_vecs[j])
-                elif ar[i][0][j] != 0:
+                elif v[0][j] == -1:
+                    ol.append(' - ' + v[1].str_vecs[j])
+                elif v[0][j] != 0:
                     # If the coefficient of the basis vector is not 1 or -1;
                     # also, we might wrap it in parentheses, for readability.
-                    arg_str = printer._print(ar[i][0][j])
-                    if isinstance(ar[i][0][j], Add):
+                    arg_str = printer._print(v[0][j])
+                    if isinstance(v[0][j], Add):
                         arg_str = "(%s)" % arg_str
                     if arg_str[0] == '-':
                         arg_str = arg_str[1:]
                         str_start = ' - '
                     else:
                         str_start = ' + '
-                    ol.append(str_start + arg_str + '*' + ar[i][1].str_vecs[j])
+                    ol.append(str_start + arg_str + '*' + v[1].str_vecs[j])
         outstr = ''.join(ol)
         if outstr.startswith(' + '):
             outstr = outstr[3:]

--- a/sympy/printing/str.py
+++ b/sympy/printing/str.py
@@ -27,6 +27,7 @@ class StrPrinter(Printer):
         "perm_cyclic": True,
         "min": None,
         "max": None,
+        "dps" : None
     }
 
     _relationals: dict[str, str] = {}
@@ -736,10 +737,9 @@ class StrPrinter(Printer):
 
     def _print_Float(self, expr):
         prec = expr._prec
-        if prec < 5:
-            dps = 0
-        else:
-            dps = prec_to_dps(expr._prec)
+        dps = self._settings.get('dps', None)
+        if dps is None:
+            dps = 0 if prec < 5 else prec_to_dps(expr._prec)
         if self._settings["full_prec"] is True:
             strip = False
         elif self._settings["full_prec"] is False:
@@ -1006,7 +1006,6 @@ class StrReprPrinter(StrPrinter):
     def _print_Str(self, s):
         # Str does not to be printed same as str here
         return "%s(%s)" % (s.__class__.__name__, self._print(s.name))
-
 
 @print_function(StrReprPrinter)
 def sstrrepr(expr, **settings):

--- a/sympy/printing/tests/test_str.py
+++ b/sympy/printing/tests/test_str.py
@@ -918,6 +918,11 @@ def test_empty_printer():
     assert str_printer.emptyPrinter(x*y) == "x*y"
     assert str_printer.emptyPrinter(32) == "32"
 
+def test_decimal_printer():
+    dec_printer = StrPrinter(settings={"dps":3})
+    f = Function('f')
+    assert dec_printer.doprint(f(1.329294)) == "f(1.33)"
+
 
 def test_settings():
     raises(TypeError, lambda: sstr(S(4), method="garbage"))

--- a/sympy/simplify/simplify.py
+++ b/sympy/simplify/simplify.py
@@ -813,7 +813,6 @@ def sum_combine(s_t):
 
     return result
 
-
 def factor_sum(self, limits=None, radical=False, clear=False, fraction=False, sign=True):
     """Return Sum with constant factors extracted.
 
@@ -832,6 +831,7 @@ def factor_sum(self, limits=None, radical=False, clear=False, fraction=False, si
     >>> factor_sum(s.function, s.limits)
     y*Sum(x, (x, 1, 3))
     """
+
     # XXX deprecate in favor of direct call to factor_terms
     kwargs = {"radical": radical, "clear": clear,
         "fraction": fraction, "sign": sign}

